pydsa/bsf.py

```diff
@@ -0,0 +1,48 @@
+"""
+
+	Breadth-first search (BFS) is an algorithm 
+	for traversing or searching tree or graph 
+	data structures. It starts at the tree root 
+	(or some arbitrary node of a graph, sometimes 
+	referred to as a 'search key'[1]) and explores 
+	the neighbor nodes first, before moving to 
+	the next level neighbors.
+
+	To find the Breadth First Search traversal
+	order of a graph given in the form of 
+	Adjacency Lists.
+	Graph is traversed from the vertex k.
+
+"""
+
+def bsf(graph,k):
+	# find number of nodes in the graph
+	n = len(graph)
+
+	# mark every node unvisited
+	visited = [False for i in range(1,n+1)]
```

Comment:
Please note that, graph can contain any object, not just integers. In that case it would not work.